src/plugins/visualizations/public/wizard/type_selection/type_selection.tsx

```diff
@@ -182,21 +182,19 @@ class TypeSelection extends React.Component<TypeSelectionProps, TypeSelectionSta
   }
 
   private filteredVisTypes(visTypes: TypesStart, query: string): VisTypeListEntry[] {
-    const types = visTypes.all().filter((type) => {
-      // Filter out all lab visualizations if lab mode is not enabled
+    const filterExperimental = (type: VisType | VisTypeAlias): boolean => {
       if (!this.props.showExperimental && type.stage === 'experimental') {
         return false;
       }
-
-      // Filter out hidden visualizations
-      if (type.hidden) {
-        return false;
-      }
-
       return true;
-    });
+    };
 
-    const allTypes = [...types, ...visTypes.getAliases()];
+    const types = visTypes
+      .all()
+      .filter(filterExperimental)
+      .filter((type) => !type.hidden); // Filter out hidden visualizations
+    const aliasedTypes = visTypes.getAliases().filter(filterExperimental);
```

Comment:
this is the important change, the rest is just refactoring the existing logic, right? In any case a unit test would make me feel a bit better about updating this...

Reply:
Yeah, but even this does not really change anything. I only abstracted the filter experimental check into its own function and applied that to aliased types too. Id love to write a test but this is a part of the `TypeSelection` component, and i'd have to write a test for the whole component to test just this change. Given what the change was, I feel fairly confident with the change. But to your point yes, ideally id like to write a test for this and the whole component too.